src/psyclone/psyir/transformations/profile_trans.py

```diff
@@ -75,3 +78,27 @@ class ProfileTrans(PSyDataTrans):
 
     def __init__(self):
         super().__init__(ProfileNode)
+
+    def validate(self, nodes, options=None):
+        """
+        Checks that the supplied list of nodes is valid for profiling
+        callipers.
+        :param nodes: a node or list of nodes to be instrumented with
+                      profiling.
+        :type nodes: :py:class:`psyclone.psyir.nodes.Node` or
+                     list[:py:class:`psyclone.psyir.nodes.Node`]
+        :raises TransformationError: if the supplied region contains an
+                                     EXIT statement.
+        """
+        # Find all the codeblocks.
+        node_list = self.get_node_list(nodes)
+        for node in node_list:
+            codeblocks = node.walk(CodeBlock)
+            for block in codeblocks:
+                if isinstance(block._fp2_nodes[0], Exit_Stmt):
```

Comment:
Would it be safer to just exclude any region with a Codeblock with:
```
excluded_node_types = (Return, Codeblock)
```
Or is this too strict?

In addition to EXIT can think of at least GOTO, ENTRY and any labelled statement that means control flow jumps into the middle or to the outside of this region and therefore miss one of the surrounding hooks.

Reply:
Update to handle GOTO and labelled though. Routines containing an ENTRY already just result in a codeblock so can't be profiled anyway.